lib/ui/painting.dart

```diff
@@ -1418,6 +1418,44 @@ class Canvas extends NativeFieldWrapperClass2 {
   /// entire group can be made transparent using the [saveLayer]'s paint.
   ///
   /// Call [restore] to pop the save stack and apply the paint to the group.
+  ///
+  /// ## Using saveLayer with clips
+  ///
+  /// When a rectanglular clip operation (from [clipRect]) is not axis-aligned
+  /// with the display, or when the clip operation is not rectalinear (e.g.
+  /// because it is a rounded rectangle clip created by [clipRRect] or an
+  /// arbitrarily complicated path clip created by [clipPath]), the edge of the
+  /// clip needs to be anti-aliased.
+  ///
+  /// If two draw calls overlap at the edge of such a clipped region, without
+  /// using [saveLayer], the first drawing will be anti-aliased with the
+  /// background first, and then the second will be anti-aliased with the result
+  /// of blending the first drawing and the background. On the other hand, if
+  /// [saveLayer] is used immediately after establishing the clip, the second
```

Comment:
but if that's the case, shouldn't that mean you don't get the problem ever? Is it just clips that are at the start of the display list that work out, and clips later in the display list are hosed? Or?

I don't understand why "draw clip draw draw" would fail but "clip draw draw" would not. why are the two clips in "clip draw clip draw" handled differently?